images/win/scripts/Installers/Install-Firefox.ps1

```diff
@@ -4,17 +4,66 @@
 ################################################################################
 
 Import-Module -Name ImageHelpers -Force
-
 $temp_install_dir = 'C:\Windows\Installer'
 New-Item -Path $temp_install_dir -ItemType Directory -Force
 
-Install-EXE -Url "https://seleniumwebdrivers.blob.core.windows.net/knownfirefoxversion/FirefoxSetup.exe" -Name "FirefoxSetup.exe" -ArgumentList "-ms"
+$versionsJson = Invoke-WebRequest  "https://product-details.mozilla.org/1.0/firefox_versions.json" | ConvertFrom-Json
+$latestVersion = $($versionsJson.LATEST_FIREFOX_VERSION)
+Write-Host "Firefox latest version: $latestVersion"
+
+# url for latest version of firefox
+$urlLatestVersion = "https://download.mozilla.org/?product=firefox-latest&os=win64&lang=en-US"
+Install-EXE -Url $urlLatestVersion -Name "Firefox Setup $latestVersion.exe" -ArgumentList ("/silent", "/install")
 
-$path = '{0}\Program Files\Mozilla Firefox\' -f $env:SystemDrive;
-New-Item -path $path -Name 'mozilla.cfg' -Value '//
+# Disable autoupdate
+$firefoxDirectoryPath = '{0}\Program Files\Mozilla Firefox\' -f $env:SystemDrive;
+New-Item -path $firefoxDirectoryPath -Name 'mozilla.cfg' -Value '//
 pref("browser.shell.checkDefaultBrowser", false);
 pref("app.update.enabled", false);' -ItemType file -force
 
-$path = '{0}\Program Files\Mozilla Firefox\defaults\pref\' -f $env:SystemDrive;
-New-Item -path $path -Name 'local-settings.js' -Value 'pref("general.config.obscure_value", 0);
+$firefoxPreferencesFolder = Join-Path $firefoxDirectoryPath "defaults\pref"
+New-Item -path $firefoxPreferencesFolder -Name 'local-settings.js' -Value 'pref("general.config.obscure_value", 0);
 pref("general.config.filename", "mozilla.cfg");' -ItemType file -force
+
+Write-Host "Install Firefox WebDriver"
+$DestinationPath = "$($env:SystemDrive)\";
+
+$geckodriverJson = Invoke-RestMethod "https://api.github.com/repos/mozilla/geckodriver/releases/latest"
+$geckodriverWindowsAsset = $geckodriverJson.assets | Where-Object { $_.name -Match "win64" } | Select-Object -First 1
+
+Write-Host "Geckodriver version: $($geckodriverJson.tag_name)"
+
+
+$DriversZipFile = $geckodriverWindowsAsset.name
+Write-Host "Selenium drivers download and install...";
```

Comment:
Please go through file and fix styling:
1) Some lines contains `;` at the end. Please get rid of them.
2) Somewhere you put 2 blank lines, somewhere 1 blank line. Use the same approach everywhere

Reply:
Done.